src/snowflake/connector/result_batch.py

```diff
@@ -304,6 +304,34 @@ def create_iter(
         """
         raise NotImplementedError()
 
+    def check_can_use_pandas(self):
```

Comment:
Took a closer look and noticed that this function in `result_batch` should use `Error.errorhandler_make_exception` instead of the Error function used in `cursor.py` (`Error.errorhandler_wrapper`). 
So, I changed the code of this function and though it is still similar, putting them in one place wouldn't work as well as it would.